### ISSUE TITLE
gcc fails on latest master

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#754]
for example: https://github.com/haproxy/haproxy/runs/873761873

```
   CC      src/backend.o
src/backend.c: In function ‘connect_server’:
src/backend.c:1363:19: error: potential null pointer dereference [-Werror=null-dereference]
 1363 |   srv_conn->owner = s->sess;
      |   ~~~~~~~~~~~~~~~~^~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:870: src/backend.o] Error 1
```

**Comment by capflam** [comment]
good catch. I forgot to catch the allocation error on `conn_new()`. I'll fix it.

**Comment by capflam** [comment]
Willy pushed the fix to 2.3-dev (dc2ac81c)

**Comment by wtarreau** [comment]
Ah sorry, I met the problem before seeing the report so I didn't mention it :-/

**Comment by wtarreau** [comment]
I'm removing the 2.2 and 2.1 tags because the commits having caused that were only merged into 2.3. I'm closing, but Ilya feel free to reopen in case the issue persists for you (not the case here at least).
